src/sentry/search/snuba/executors.py

```diff
@@ -619,4 +627,204 @@ def calculate_hits(
+        # https://github.com/getsentry/sentry/blob/804c85100d0003cfdda91701911f21ed5f66f67c/src/sentry/event_manager.py#L241-L271
+        "priority": Function(
+            "toUInt64",
+            [
+                Function(
+                    "plus",
+                    [
+                        Function(
+                            "multiply",
+                            [
+                                Function(
+                                    "log",
+                                    [times_seen_aggregation],
+                                ),
+                                600,
+                            ],
+                        ),
+                        last_seen_aggregation,
+                    ],
+                )
+            ],
+        ),
+        "user_count": Function("uniq", [Column("tags[sentry:user]", entities["event"])]),
+    }
+
+    def __init__(self):
+        super().__init__()
+        self.postres_executor = PostgresSnubaQueryExecutor()
+
+    def calculate_start_end(
+        self,
+        retention_window_start: Optional[datetime],
+        search_filters: Sequence[SearchFilter],
+        date_from: Optional[datetime],
+        date_to: Optional[datetime],
+    ):
+        now = timezone.now()
+        end = None
+        end_params = [_f for _f in [date_to, get_search_filter(search_filters, "date", "<")] if _f]
+        if end_params:
+            end = min(end_params)
+
+        if not end:
+            end = now + ALLOWED_FUTURE_DELTA
+
+        retention_date = max(_f for _f in [retention_window_start, now - timedelta(days=90)] if _f)
+        start_params = [date_from, retention_date, get_search_filter(search_filters, "date", ">")]
+        start = max(_f for _f in start_params if _f)
+        end = max([retention_date, end])
+        return start, end, retention_date
+
+    def query(
+        self,
+        projects: Sequence[Project],
+        retention_window_start: Optional[datetime],
+        group_queryset: QuerySet,
+        environments: Sequence[Environment],
+        sort_by: str,
+        limit: int,
+        cursor: Optional[Cursor],
+        count_hits: bool,
+        paginator_options: Mapping[str, Any],
+        search_filters: Sequence[SearchFilter],
+        date_from: Optional[datetime],
+        date_to: Optional[datetime],
+        max_hits=None,
+    ) -> CursorResult:
+        # TODO: For the moment, the assumption is that this backend will only ever be passed
+        # search_filters derived from `is:unresolved`. We'll hardcode that case and handle more
+        # later.
+
+        if search_filters != [SearchFilter(SearchKey("status"), "IN", SearchValue([0]))]:
+            logging.error(
+                "Received invalid search_filters in `CdcPostgresSnubaQueryExecutor`, "
+                "passing through to base class",
+                extra={"search_filters": search_filters},
+            )
+            return self.postres_executor.query(
```

Comment:
Why do you still want to run the non-cdc executor here? I think returning an error here should be ok (for now, I guess `return self.empty_result`)? Wouldn't this just mean we're running it twice for queries we don't really care about comparing or would this skew stats in the service delegator?

I'm wondering if it's better to let the executors be "dumber", especially if we have tests that can verify the right executor is used for certain queries (lol...or _query_ for now). Leave it up to the search backend to determine what executor to use. Perhaps we can define a set supported query "rules" for each executor, and check them in the backend in order of executor priority, choosing the first one that passes all requirements. 

I just realized this would be pretty similar to how the current `PostgresSnubaQueryExecutor` chooses between just postgres or postgres+snuba...but brought into the backend class a little bit more neatly defined, and not deeply integrated in the `query` function